### DISCUSSION TITLE
undefined method `audit_log_metadata_for'

### DISCUSSION THREAD

**Discussion by francescor** [janko/rodauth-omniauth#27] [General]
When I add in `app/misc/rodauth_main.rb` (is it the right place?)

```
      audit_log_metadata_for :login do
        { "provider" => omniauth_provider } if authenticated_by.include?("omniauth")
      end
```

I get

```
  undefined method `audit_log_metadata_for' for #<Rodauth::Configuration:0x00007f6ecd635ec8 @auth=RodauthMain>
```

I see I have `rodauth-omniauth (0.5.1)` in Gemfile.lock then `ruby 2.6.9p207, Rails 5.2.8.1`

**Comment by francescor** [comment]
uh, sorry, I see it's probably because I am not using the so called `audit_logging`... 

is it a gem? this one https://github.com/rails-engine/audit-log ?

**Reply by janko** [reply]
This was referring to Rodauth's [audit_logging](https://rodauth.jeremyevans.net/rdoc/files/doc/audit_logging_rdoc.html) plugin. I just added `enable :audit_logging` in the README example to avoid confusion.
